### ISSUE TITLE
Adding secp256k1 support

### ISSUE THREAD

**Issue by infinity0** [haskell-cryptography/cacophony#7]
[lightning uses noise-with-secp256k1](https://github.com/lightningnetwork/lightning-rfc/blob/master/08-transport.md#noise-protocol-instantiation) even though this instantiation is not explicitly mentioned in the [noise protocol spec](https://noiseprotocol.org/noise.html).

Are there any plans to add this to cacophony, or would you be happy to accept a PR implementing such support?

**Comment by centromere** [comment]
I would accept a PR providing such support.

**Comment by ovangle** [comment]
Are you OK with secp256k1 from cryptonite lib? If so I could create a PR for that.

**Comment by centromere** [comment]
Yes. I am OK with that because it doesn't introduce an additional dependency.

**Comment by ysangkok** [comment]
I have been working on this. We need to support the custom key rotation mechanism they use in Lightning, but if we do, it will work. I have a PR for secp256k1 support in Cryptonite, let's see if I can get that merged. If I can't, would you be ok with using the ECDH bindings that I could add to secp256k1-haskell? That would require one new dependency then.

**Comment by centromere** [comment]
Yes.

**Comment by ysangkok** [comment]
See PR #13
